presto-tests/src/test/java/com/facebook/presto/tests/AbstractTestDistributedStats.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tests;
+
+import com.facebook.presto.execution.QueryPlan;
+import com.facebook.presto.execution.StageInfo;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.tests.statistics.Metric;
+import com.facebook.presto.tests.statistics.MetricComparator;
+import com.facebook.presto.tests.statistics.MetricComparison;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunnerWithoutCatalogs;
+import static java.util.Collections.emptyMap;
+
+public class AbstractTestDistributedStats
```

Comment:
I'd like to avoid repeating the mistake of abstract test class again. I see two options:
1. extract colaborator object
2. extract interface with default methods, inheriting a 
```
interface DistributedQueryRunnerTest {
    DistributedQueryRunner getQueryRunner()
}
```

I think 2. is viable and preferable.